PetGame/Services/PetService.cs

```diff
@@ -200,5 +211,196 @@ public bool DeletePet(ulong petId, ulong ownerId)
             // return true if exactly one pet was deleted
             return results == 1;
         }
+
+        /// <summary>
+        /// Returns a PetStatus that contains the Pet, its hunger percentage,
+        /// its happiness percentage, and the DateTime of next interaction
+        /// </summary>
+        /// <param name="PetId"></param>
+        /// <returns>
+        /// PetStatus object
+        /// happiness: 0= saddest 1 = happiest
+        /// hunger: 0= very hungry 1 = not hungry
+        /// </returns>
+        public PetStatus GetPetStatusById(ulong PetId)
+        {
+            //call PetService function to get the Pet
+            Pet toReturn = GetPetById(PetId);
+
+            //if this pet doesn't exist, don't go any further. return null
+            if (toReturn == null)
+            {
+                return null;
+            }
+
+            //hunger starts at 100%
+            double hungerPercentage = 1.0;
+
+            //happiness starts at 100%
+            double happinessPercentage = 1.0;
+
+            DateTime ToCheck = DateTime.Now.Subtract(TimeSpan.FromDays(NumDaysToCheck));
+            int HoursToCheck = NumDaysToCheck * 24;
+
+            List<Activity> PastActivities = (List<Activity>)activityService.GetActivities(PetId, NumActivities, ToCheck, null);
```

Comment:
You should not cast an `IEnumerable` into a `List`. Keep it as an `IEnumerable`.